libraries/AP_HAL/I2CDevice.h

```diff
@@ -76,6 +76,11 @@ class I2CDeviceManager {
                                                  uint32_t bus_clock=400000,
                                                  bool use_smbus = false,
                                                  uint32_t timeout_ms=4) = 0;
+
+    virtual AP_HAL::I2CDevice * get_device_direct(uint8_t bus, uint8_t address,
```

Comment:
Needs some HAL_Linux/HAL_SITL implementations. I think @peterbarker has enough going that at a minimum SITL should be added and testable these days as a blocker, before merge. Linux may be a separate debate.

Reply:
removed this funciton